cpp/src/streaming/core/spillable_messages.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <rapidsmpf/streaming/core/spillable_messages.hpp>
+
+namespace rapidsmpf::streaming {
+
+SpillableMessages::MessageId SpillableMessages::insert(Message&& message) {
+    std::lock_guard<std::mutex> lock(global_mutex_);
+    content_descriptions_.insert({counter_, message.content_description()});
+    items_.insert({counter_, std::make_shared<Item>(std::move(message))});
+    return counter_++;
+}
+
+Message SpillableMessages::extract(MessageId mid) {
+    std::unique_lock global_lock(global_mutex_);
+    std::shared_ptr<Item> item = extract_item(items_, mid).second;
+    content_descriptions_.erase(mid);
+    global_lock.unlock();
+
```

Comment:
*question*: If someone else `extract`ed this message ID, I think that `item` at this point would be `nullptr`, or have we already `throw`n because `extract_item` throws if the key does not exist?

Reply:
Yes, `extract_item()` will throw.